Creating bindings for libwinit

Apparently, Clang.jl errors when translating this bit of code:
```c
#include <stdint.h>

enum X {
  A,
  B,
  C,
};
typedef uint32_t X;
```

The error is:

```
ERROR: LoadError: AssertionError: ptr_ref[] != C_NULL
Stacktrace:
 [1] Clang.TokenList(tu::Ptr{Clang.LibClang.CXTranslationUnitImpl}, sr::Clang.LibClang.CXSourceRange)
   @ Clang ~/.julia/packages/Clang/HZdsi/src/token.jl:13
 [2] tokenize(c::Clang.CLNoDeclFound)
   @ Clang ~/.julia/packages/Clang/HZdsi/src/token.jl:105
 [3] is_same(cursor1::Clang.CLNoDeclFound, cursor2::Clang.CLEnumDecl)
   @ Clang.Generators ~/.julia/packages/Clang/HZdsi/src/generator/utils.jl:19
 [4] sanity_check(dag::ExprDAG, options::Dict{String, Any})
   @ Clang.Generators ~/.julia/packages/Clang/HZdsi/src/generator/audit.jl:68
 [5] (::Audit)(dag::ExprDAG, options::Dict{String, Any})
   @ Clang.Generators ~/.julia/packages/Clang/HZdsi/src/generator/passes.jl:859
 [6] build!(ctx::Context, stage::Clang.Generators.BuildStage)
   @ Clang.Generators ~/.julia/packages/Clang/HZdsi/src/generator/context.jl:169
 [7] build!(ctx::Context)
   @ Clang.Generators ~/.julia/packages/Clang/HZdsi/src/generator/context.jl:160
 [8] top-level scope
   @ ~/generator.jl:8
   ```
   
   This is technically valid C code.

--- a/src/generator/context.jl
+++ b/src/generator/context.jl
@@ -108,8 +108,10 @@ function create_context(headers::Vector, args::Vector=String[], options::Dict=Di
     if get(general_options, "smart_de_anonymize", true)
         push!(ctx.passes, DeAnonymize())
     end
-    if !get(general_options, "no_audit", false)
-        @error "The generator is running in `no_audit` mode. It could generate invalid Julia code. Please DO NOT submit issues only occur in this mode. You can remove the `no_audit` entry in the `.toml` file to exit this mode."
+    if get(general_options, "no_audit", false)
+        @error "The generator is running in `no_audit` mode. It could generate invalid Julia code. You can remove the `no_audit` entry in the `.toml` file to exit this mode."
+        get(general_options, "link_enum_alias", true) && push!(ctx.passes, LinkEnumAlias())
+    else
         push!(ctx.passes, Audit())
     end
     push!(ctx.passes, Codegen())

--- a/src/generator/passes.jl
+++ b/src/generator/passes.jl
@@ -507,6 +507,41 @@ function (x::CatchDuplicatedAnonymousTags)(dag::ExprDAG, options::Dict)
 end
 
 """
+    LinkEnumAlias <: AbstractPass
+Link hard-coded enum types to the corresponding enums. This pass only works in `no_audit` mode.
+"""
+mutable struct LinkEnumAlias <: AbstractPass
+    show_info::Bool
+end
+LinkEnumAlias(; info=false) = LinkEnumAlias(info)
+
+function (x::LinkEnumAlias)(dag::ExprDAG, options::Dict)
+    general_options = get(options, "general", Dict())
+    log_options = get(general_options, "log", Dict())
+    show_info = get(log_options, "LinkEnumAlias_log", x.show_info)
+
+    for (id,i) in dag.ids
+        node = dag.nodes[i]
+        node.type isa AbstractTypedefNodeType || continue
+
+        ty = getTypedefDeclUnderlyingType(node.cursor) |> getCanonicalType
+        typeKind = kind(ty)
+        typeKind == CXType_Int || typeKind == CXType_Short || typeKind == CXType_Long || typeKind == CXType_LongLong ||
+        typeKind == CXType_UInt || typeKind == CXType_UShort || typeKind == CXType_ULong || typeKind == CXType_ULongLong ||
+        continue
+
+        for (tagid, j) in dag.tags
+            dag.nodes[j].type isa AbstractEnumNodeType || continue
+            tagid == id || continue
+
+            dag.nodes[i] = ExprNode(id, SoftSkip(), node.cursor, node.exprs, node.adj)
+            show_info &&
+                @info "[LinkEnumAlias]: skip $id at dag.nodes[$i]."
+        end
+    end
+end
+
+"""
     DeAnonymize <: AbstractPass
 In this pass, naive anonymous tag-types are de-anonymized and the correspoding typedefs
 are marked [`Skip`](@ref).

--- a/test/generators.jl
+++ b/test/generators.jl
@@ -153,3 +153,9 @@ end
     ctx = create_context([joinpath(@__DIR__, "include/enum.h")], get_default_args())
     @test_throws Exception build!(ctx)
 end
+
+@testset "Issue 412 - no audit" begin
+    options = Dict("general" => Dict{String,Any}("no_audit" => true))
+    ctx = create_context([joinpath(@__DIR__, "include/enum.h")], get_default_args(), options)
+    @test_logs (:info, "Done!") match_mode = :any build!(ctx)
+end

